base_station/gui/src/components/SASiteControls.vue

```diff
@@ -89,11 +91,155 @@
           'type': 'MicroCam',
           'id': 'camera_' + this.site
         })
+      },
+      swapComponent: function(component)
+      {
+        this.currentComponent = component;
       }
     },
 
     components: {
-      Checkbox
+      Checkbox,
+      'Chlorophyll': {
+        template: '<div><p>Turn off White LEDs</p><button id="execute" @click="swapComponent(\'WhiteLEDsOff\')">Execute</button></div>',
+        methods: {
+          swapComponent: function(component)
+          {
+            this.$parent.$parent.$parent.publish("/mosfet_cmd", {
+              'type': 'MosfetCmd',
+              'device': 2,
+              'enable': false
+            })
+            var site_letter = 'b';
+            if (this.$parent.site == "White") {
+              site_letter = 'w';
+            }
+            else if (this.$parent.site == "Yellow") {
+              site_letter = 'y';
+            }
+            // alert(site_letter);
+            this.$parent.$parent.$parent.publish("/spectral_cmd", {
+              'type': 'SpectralCmd',
+              'sensor': site_letter
```

Comment:
we dont send the site letter, idk if teleop calculates this but we just send a struct with 18 chanels of data, 6 per sensor, if we don't get data from one spectral sensor the channel data is all zeros

Reply:
This file was removed